src/dataflow/src/render/context.rs

```diff
@@ -65,6 +71,23 @@ where
     T: Timestamp + Lattice,
     S::Timestamp: Lattice + Refines<T>,
 {
+    /// The debug name of the dataflow associated with this context.
+    pub debug_name: String,
+    /// When set, `as_of` indicates a frontier that can be used to compact input
+    /// timestamps without affecting the results. We *should* apply it, to
+    /// sources and imported traces, both because it improves performance, and
+    /// because potentially incorrect results are visible in sinks.
+    pub as_of_frontier: Antichain<dataflow_types::Timestamp>,
+    /// The source tokens for all sources that have been built in this context.
+    pub source_tokens: HashMap<GlobalId, Rc<Option<SourceToken>>>,
+    /// The index tokens for all indexes that have been built in this context.
+    pub index_tokens: HashMap<GlobalId, Rc<dyn Any>>,
+    /// A hacky identifier for the DataflowDesc associated with this context.
+    ///
+    /// This is stopgap measure so dropping an index and recreating one with the
+    /// same name does not result in timestamp/reading from source errors. Use
+    /// an export id to distinguish between different dataflows.
+    pub first_export_id: GlobalId,
```

Comment:
Do we have an issue filed for this? Also, just reading through the diffs and seeing things that are not your fault, so feel free to punt (but seemed like a good time to ask!).

Reply:
Colocated the two comments!